File: remove_duplicates_from_sorted_array.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not isinstance(nums, list):
            raise TypeError('nums must be a valid list object that must contains int object type elements')
        
        if len(nums) == 0:
            return 0 
        
        left = 0
        right = len(nums) - 1
        
        while left < right:
            if nums[left] == nums[right]:
                nums.remove(nums[left])
            
            right -= 1  
        
        return len(nums)
            
obj = Solution()
print(obj.removeDuplicates([1,1,2]))
print(obj.removeDuplicates([0,0,1,1,1,2,2,3,3,4])) 

File: same_solution_with_different_language/remove_duplicates_from_sorted_array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
    if (!Array.isArray(nums)) {
        throw new TypeError('nums must be a valid Array');
    }
    else if (nums.length == 0) {
        return null;
    }

    let left = 0;
    let right = nums.length;

    while (left < right) {
        if (nums[left] == nums[right]) {
            nums.removeDuplicates()
        }
    }
};
console.log(removeDuplicates([1, 1, 2]));
console.log(removeDuplicates([0, 0, 1, 1, 1, 2, 2, 3, 3, 4]));